src/lib/homestore_backend/hs_pg_manager.cpp

```diff
@@ -193,4 +206,73 @@ void HSHomeObject::persist_pg_sb() {
     }
 }
 
+bool HSHomeObject::_get_stats(pg_id_t id, PGStats& stats) const {
+    auto lg = std::shared_lock(_pg_lock);
+    auto it = _pg_map.find(id);
+    if (_pg_map.end() == it) { return false; }
+    auto const& pg = it->second;
+    auto hs_pg = static_cast< HS_PG* >(pg.get());
+    auto const blk_size = hs_pg->repl_dev_->get_blk_size();
+
+    stats.id = hs_pg->pg_info_.id;
+    stats.replica_set_uuid = hs_pg->pg_info_.replica_set_uuid;
+    stats.num_members = hs_pg->pg_info_.members.size();
+    stats.total_shards = hs_pg->total_shards();
+    stats.open_shards = hs_pg->open_shards();
+
+    for (auto const& m : hs_pg->pg_info_.members) {
```

Comment:
the quorum in PGStat need `up to date PG members`.  The initiative of reporting this to CM is let client (GW) "best effort" avoid read send to a "significant laggy" member,  especially , those members are under snapshot based resync.

We can only get the member stats from leader so we may only report this in leader PG. 
Also the `Quorum` struct need leader information which is not in pg_stat :)

Reply:
good catch, it should be inside HomeObj stats.

Reply:
as raft part is not yet merged, maybe we merge this PR first and create an issue for the PG member change?

Reply:
yes, I added the required fields into the stats and left that field as zero for now. I will create another ticket to track this issue once repl_dev is ready to get the api for last committed lsn.